File: C++/Medium/WrongAnswer/maximum-xor-after-operations_0.cpp
class Solution {
public:
    int maximumXOR(vector<int>& nums) {
        
        int ans = 0;
        int mask = 0;
        for (int i = 30; i >= 0; i--) {
            mask |= (1 << i);
            unordered_set<int> s;
            for (int num : nums) {
                s.insert(num & mask);
            }
            int tmp = ans | (1 << i);
            for (int prefix : s) {
                if (s.find(prefix ^ tmp) != s.end()) {
                    ans = tmp;
                    break;
                }
            }
        }
        return ans;
    }
};

File: Rust/Medium/WrongAnswer/maximum-xor-after-operations_0.rs
impl Solution {
    pub fn maximum_xor(nums: Vec<i32>) -> i32 {
        
        let mut ans = 0;
        let mut mask = 0;
        for i in (0..=30).rev() {
            mask |= 1 << i;
            let mut s = std::collections::HashSet::new();
            for n in nums.iter() {
                s.insert(n & mask);
            }
            let cand = ans | 1 << i;
            for p in s.iter() {
                if s.contains(&(p ^ cand)) {
                    ans = cand;
                    break;
                }
            }
        }
        ans
    }
}